src/charm.py

```diff
@@ -282,6 +238,18 @@ def _on_database_relation_broken(self, event: EventBase) -> None:
         """
         self.unit.status = BlockedStatus("Waiting for database relation")
 
+    def _certificates_updated(self) -> bool:
```

Comment:
This method seems to do 2 things at once, it returns whether the existing cert matches the csr plus it stores the certificate, I'd encourage not to have those 2 behaviors in this same method and have it only return a bool. The method name does not suggest that it affects anything

Reply:
No, it scans all the results from the library, and if one matches the current CSR, then it writes the config if it is different.  Essentially it is a check to see if the cert was updated for any reason.